cmd/node/main.go

```diff
@@ -1481,6 +1562,35 @@ func createNodesCoordinator(
 	return nodesCoordinator, nil
 }
 
+func serializableValidatorsToValidators(nodeRegistryValidators map[string][]*sharding.SerializableValidator) (map[uint32][]sharding.Validator, error) {
+	validators := make(map[uint32][]sharding.Validator)
+	for shardId, shardValidators := range nodeRegistryValidators {
+		newValidators, err := serializableShardValidatorListToValidatorList(shardValidators)
+		if err != nil {
+			return nil, err
+		}
+		shardIdInt, err := strconv.ParseUint(shardId, 10, 32)
+		if err != nil {
+			return nil, err
+		}
+		validators[uint32(shardIdInt)] = newValidators
+	}
+
+	return validators, nil
+}
+
+func serializableShardValidatorListToValidatorList(shardValidators []*sharding.SerializableValidator) ([]sharding.Validator, error) {
```

Comment:
make the methods in the sharding folder public - no need in main

Reply:
fixed